vertx/src/main/java/com/microsoft/gctoolkit/vertx/jvm/AbstractJavaVirtualMachine.java

```diff
@@ -32,95 +29,112 @@
+    private Diary diary;
     private DateTimeStamp timeOfLastEvent;
     private final Map<Class<? extends Aggregation>, Aggregation> aggregatedData = new ConcurrentHashMap<>();
 
     @Override
     public boolean isG1GC() {
-        return jvmConfigurationFromParser.getDiary().isG1GC();
+        return diary.isG1GC();
     }
 
     @Override
     public boolean isZGC() {
-        return jvmConfigurationFromParser.getDiary().isZGC();
+        return diary.isZGC();
     }
 
     @Override
     public boolean isShenandoah() {
-        return jvmConfigurationFromParser.getDiary().isShenandoah();
+        return diary.isShenandoah();
     }
 
     @Override
     public boolean isParallel() {
-        return jvmConfigurationFromParser.getDiary().isParNew();
+        return diary.isParNew();
     }
 
     @Override
     public boolean isSerial() {
-        return jvmConfigurationFromParser.getDiary().isSerialFull();
+        return diary.isSerialFull();
     }
 
     @Override
     public boolean isCMS() {
-        return jvmConfigurationFromParser.getDiary().isCMS();
+        return diary.isCMS();
     }
 
     @Override
     public String getCommandLine() {
-        return jvmConfigurationFromParser.getCommandLine();
+        return ""; //todo: extract from diary... jvmConfigurationFromParser.getCommandLine();
     }
 
+    /**
+     * of the first event is significantly away from zero in relation to the time intervals between the
+     * of the next N events, where N maybe 1.
+     *
+     * try to estimate the time at which the JVM started. For log fragments, this will be the time
+     * of the first event in the log. Otherwise it will be 0.000 seconds.
+     * @return DateTimeStamp
+     */
+    @Override
+    public DateTimeStamp getEstimatedJVMStartTime() {
+        DateTimeStamp startTime = getTimeOfFirstEvent();
+        // Initial entries in GC log happen within seconds. Lets allow for 60 before considering the log
+        // to be a fragment.
+        if (startTime.getTimeStamp() < LOG_FRAGMENT_THRESHOLD) {
+            return startTime.minus(startTime.getTimeStamp());
+        } else {
+            return startTime;
+        }
+    }
+
+    /**
+     * todo: fix this to be a globally available value. Also, the JVM start time is not zero if the time
+     * of the first event is significantly away from zero in relation to the time intervals between the
+     * of the next N events, where N maybe 1.
+     *
+     * try to estimate the time at which the JVM started. For log fragments, this will be the time
+     * of the first event in the log. Otherwise it will be 0.000 seconds.
+     * @return DateTimeStamp
+     */
     @Override
     public DateTimeStamp getTimeOfFirstEvent() {
-        return jvmConfigurationFromParser.getTimeOfFirstEvent();
+        return diary.getTimeOfFirstEvent();
     }
 
+    /**
+     * JVM termination time will be one of either, the time stamp in the termination event if present or, the
```

Comment:
I think I mentioned elsewhere that it isn't correct to call this JVM termination when it is really the end of the log file(s).

Reply:
For the purposes of the analysis, I think it's ok to do this. Otherwise we'd have to add and EOL event.